tools/shared.py

```diff
@@ -3057,13 +3057,52 @@ def safe_copy(src, dst):
     return
   shutil.copyfile(src, dst)
 
+def find_lines_with_preprocessor_directives(file):
+  return filter(lambda x: '#if' in x or '#elif' in x, open(file, 'r').readlines())
+
+def run_c_preprocessor_on_file(src, dst):
```

Comment:
I think this clang preprocessor change seems unnecessary in this PR? (Also, I was hoping we don't need this after the recent preprocessor eval improvement?)

Reply:
Migrated to use the newer preprocessor evaling code